modules/http4s/src/smithy4s/http4s/SmithyHttp4sRouter.scala

```diff
@@ -35,7 +34,7 @@ class SmithyHttp4sRouter[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _], F[_]](
   val routes: HttpRoutes[F] = Kleisli { request =>
     for {
       endpoints <- perMethodEndpoint.get(request.method).toOptionT[F]
-      path = matchPath.make(request.uri.path.renderString)
+      path = request.uri.path.segments.map(_.decoded()).toArray
```

Comment:
I think I'll extract this part to a function and unit test that with some strange cases (uriencoded segments, trailing/leading slashes) but the bulk of the work is on the http4s side here

Reply:
I'd be keen to know what kind of primitives we get from other test frameworks too (akka/play), which we support internally, before we decide to tag this as deprecated [this](https://github.com/disneystreaming/smithy4s/pull/284/files#r907541321)

Reply:
Play doesn't seem to expose segments so we'll probably have to keep that for a while